app/lib/posts/update.rb

```diff
@@ -0,0 +1,61 @@
+module Posts
+  class Update < Base
+    attr_reader :post, :title, :status, :user_id, :content
+
+    def validate!
+      validate_user_has_permission
+    end
+
+    def run!
+      update_post
+    end
+
+    private
+
+    def update_post
+      @post.title = @title if @title
+      if @status
+        log_activity
+        @post.status = @status
+      end
+      if @user_id
+        @post.user_id = @user_id
+        add_voter
+      end
+      @post.content.body = @content["body"] if @content && @content["body"]
+      @post.slug = nil if @post.title_changed?
+
+      @post.save!
+    end
+
+    def log_activity
+      ActiveRecord::Base.transaction do
```

Comment:
We should only log activity if the status is being changed. We should probably add a test for this!

```suggestion
    def log_activity
      return if @post.status == @status
    
      ActiveRecord::Base.transaction do
```